client/client.go

```diff
@@ -32,11 +32,57 @@ func New(apiEndpoint string) *KalaClient {
 	if strings.HasSuffix(apiEndpoint, "/") {
 		apiEndpoint = apiEndpoint[:len(apiEndpoint)-1]
 	}
+	apiUrlPrefix := api.ApiUrlPrefix[:len(api.ApiUrlPrefix)-1]
 
 	return &KalaClient{
-		apiEndpoint: apiEndpoint + api.ApiUrlPrefix,
-		requester:   sling.New().Base(apiEndpoint + api.ApiUrlPrefix),
+		apiEndpoint: apiEndpoint + apiUrlPrefix,
+	}
+}
+
+func (kc *KalaClient) encode(value interface{}) (io.Reader, error) {
+	if value == nil {
+		return nil, nil
+	}
+	buf := new(bytes.Buffer)
+	if err := json.NewEncoder(buf).Encode(value); err != nil {
+		return nil, err
+	}
+	return buf, nil
+}
+
+func (kc *KalaClient) decode(body io.Reader, target interface{}) error {
+	if target == nil {
+		return nil
+	}
+	return json.NewDecoder(body).Decode(target)
+}
+
+func (kc *KalaClient) url(parts ...string) string {
+	for i, part := range parts {
+		if part == "" {
+			continue
+		}
+		if init, last := part[:len(part)-1], part[len(part)-1]; last == '/' {
+			parts[i] = init
+		}
+	}
+	return strings.Join(append([]string{kc.apiEndpoint}, parts...), "/") + "/"
+}
```

Comment:
This method's logic seems too tricky. I believe `path.Join` would suffice: https://golang.org/pkg/path/#Join